### PR TITLE
Company-Creation-and-Management-(Affiliate-+-Supplier)

### PR DESCRIPTION
**Pull request by h0bbitten**

### CODE REVIEW

**Comment by tblank23** [comment]
Maybe change "affiliates" to be singular @frederikgustav do you agree?

**Comment by h0bbitten** [comment]
I give up

File: app/models/company.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Company < ActiveRecord::Base
+  validates :name, presence: true
+  validates :id, presence: true, uniqueness: true
```
Comment (soupglasses) [inline app/models/company.rb:3]: Is already done in the DB. Could be omitted.

File: db/migrate/20241125095123_create_companies.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class CreateCompanies < ActiveRecord::Migration[8.0]
+  def change
+    create_table :companies do |t|
+      t.string :name
+      t.integer :company_id
```
Comment (soupglasses) [inline db/migrate/20241125095123_create_companies.rb:5]: What is this used for?

**Comment by frederikgustav** [comment]
> Maybe change "affiliates" to be singular @frederikgustav do you agree?

I definitely do, i think some of the validations are sorted out on db level too, so those can be removed, @tblank23 do you agree?

**Comment by tblank23** [comment]
> > Maybe change "affiliates" to be singular @frederikgustav do you agree?
> 
> I definitely do, i think some of the validations are sorted out on db level too, so those can be removed, @tblank23 do you agree?

Yeah I believe that would be in order, unless you want to remove that check from the database and implement it here instead? Any thoughts @frederikgustav

**Comment by frederikgustav** [comment]
> > > Maybe change "affiliates" to be singular @frederikgustav do you agree?
> > 
> > 
> > I definitely do, i think some of the validations are sorted out on db level too, so those can be removed, @tblank23 do you agree?
> 
> Yeah I believe that would be in order, unless you want to remove that check from the database and implement it here instead? Any thoughts @frederikgustav
No further comments...

File: db/migrate/20241125095123_create_companies.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class CreateCompanies < ActiveRecord::Migration[8.0]
+  def change
+    create_table :companies do |t|
```
Comment (soupglasses) [inline db/migrate/20241125095123_create_companies.rb:3]: Add `t.type :string`, as the inheritance is stored in the companies table.

File: db/migrate/20241125095656_create_suppliers.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class CreateSuppliers < ActiveRecord::Migration[8.0]
+  def change
+    create_table :suppliers do |t|
+      t.timestamps
+    end
+  end
+end
```
Comment (soupglasses) [inline db/migrate/20241125095656_create_suppliers.rb:7]: Remove

File: db/migrate/20241125095745_create_affiliates.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class CreateAffiliates < ActiveRecord::Migration[8.0]
+  def change
+    create_table :affiliates do |t|
+      t.timestamps
+    end
+  end
+end
```
Comment (soupglasses) [inline db/migrate/20241125095745_create_affiliates.rb:7]: Remove

**Review by soupglasses — CHANGES_REQUESTED** [review]

File: db/migrate/20241125095123_create_companies.rb (hunk 2 above)
Comment (h0bbitten) [reply db/migrate/20241125095123_create_companies.rb:5]: Well attributes?

**Review by soupglasses — APPROVED** [review]
lgtm

**Pull request merged**